### ISSUE TITLE
Feature: Add option to disable any external links/lookups/connections

### ISSUE THREAD

**Issue by eharris** [airsonic/airsonic#712]
I would like to install an instance of airsonic on a private network that for security reasons does not allow outside internet access.  Any attempt to connect to outside services is considered a security issue.  For this reason, I'd need to be able to completely disable the server from ever attempting to connect to any other outside host or service, it would only be for local private network access.

This would include the need to disable the ability to check for updates, look up cover art, connect to last.fm, google, wikipedia, allmusic, etc.  Even dns lookups might be a problem as they could just hang with no response.

I know that the user administration allows for users to be restricted from being able to change the cover art or share files, but I don't see any way to disable the links to outside search services.  And there should be a way to ensure that nothing else the server does under the covers will ever result in a connection attempt or a link that might be followed to an off-net service.

**Comment by jooola** [comment]
Yes I think we already had some similar ticket asking to disable last.fm artist info fetch and other.

As work around I suggest you to add those domains into your hosts file and block them (redirect to 127.0.0.1)

Still I don't know what are all the external services that airsonic uses so you can try those and test. You could also try to monitor yo out trafic and see if they are any other services.

From what I know : 
- Last FM Scrobble `post.audioscrobbler.com`
- Last FM `last.fm`
- Chart Lyrics `api.chartlyrics.com`

Unsure :
- Podcasts `itunes.com`
- Sonos `sonos.com`

I would be surprised if those are used at all but adding them here anyway :
- System `www.w3.org`
- System `www.springframework.org`
- System `www.liquibase.org`
- System `www.macromedia.com`

Same for those : 
```
<%@ page session="false"%>
<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
<%@ taglib prefix="sec" uri="http://www.springframework.org/security/tags" %>
<%@ taglib prefix="sub" uri="http://airsonic.org/taglib/sub" %>
<%@ taglib prefix="str" uri="http://jakarta.apache.org/taglibs/string-1.1" %>
```
Maybe they are only fetched on client side ?
Like Google fonts.

**Comment by eharris** [comment]
This points out another issue, which is if there are javascript, font, or other fetches from outside sites, that would also be a problem, and it would be more of an issue than just a security concern, it could also make part or all of the system non-functional.  Everything the server depends on would need to be self-contained.

**Comment by jooola** [comment]
@eharris Yup mainly about the fonts I already wanted to add them locally but haven't had time to do it.
Still they are fall back fonts anyway. 

But are the clients also in this network ? I though the client where outside the network and so fetching fonts and JS stuff should not be a problem.

**Comment by eharris** [comment]
No, the clients will also be on the private network, that is why airsonic can't serve any links that might be accidentally clicked that would cause clients to try to load external resources.

**Comment by jvoisin** [comment]
A workaround is to use a CSP policy to prevent the browser from downloading external resources, like this one:

```
default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; object-src 'none'
```

**Comment by jvoisin** [comment]
This might be a stupid question, but do we even need custom fonts?

This is how my instance is looking without without any google font:

![Screenshot_2019-03-21_19-46-41](https://user-images.githubusercontent.com/325724/54777572-7baf9800-4c0a-11e9-8537-2db7b5427ebc.png)

**Comment by jvoisin** [comment]
It seems that the only client-side dependencies are now recaptcha and the [chromecast thingy]( https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1 ). The former can easily be replaced by an other self-hosted captcha package, and the later is a few lines of javascript that could be embedded in airsonic.

**Comment by eharris** [comment]
I don't think the issue with the airsonic version check has been addressed yet.  I mentioned this also no the PR.

**Comment by eharris** [comment]
@jvoisin Re: your comment about captcha, I haven't looked into it that deeply, but I believe that the captcha can that already be disabled on the `Advanced` settings page, which I think is sufficient.

So I think the only remaining issues are:

* The airsonic version check
* The chromecast support

Anything else I'm missing?

**Comment by jvoisin** [comment]
Nope, this should be all. Worse case, we can always pile more things on top of it afterwards.

**Comment by dud1337** [comment]
This issue appears to have been brought up multiple times in the past and on the reddit. Would really prefer at least the option to disable it.

**Comment by eharris** [comment]
For easy reference, the PR to address Google Chrome Cast is #1281

**Comment by Perflyst** [comment]
@jvoisin with your CSP it is no longer possible to open the settings, sadly I did not found a solution
